pallets/liquid-staking/src/benchmarking.rs

```diff
@@ -0,0 +1,251 @@
+            INITIAL_AMOUNT - amount
+        );
+
+        assert_eq!(
+            T::Currency::free_balance(T::StakingCurrency::get(), &agent),
+            withdraw_amount
+        );
+
+        assert_last_event::<T>(Event::<T>::WithdrawSuccess(agent, withdraw_amount).into());
+    }
+
+    record_rewards {
+        let caller: T::AccountId = whitelisted_caller();
+        initial_set_up::<T>(caller.clone());
+        let agent: T::AccountId = account("Sample", 6, SEED);
+        let amount = 100_000;
+        assert_ok!(LiquidStaking::<T>::stake(
+            SystemOrigin::Signed(caller.clone()).into(),
+            amount));
+        let call = Call::<T>::record_rewards(agent.clone(), amount);
+        let origin = SystemOrigin::Root.into();
+    }: { call.dispatch_bypass_filter(origin)? }
+    verify {
+        assert_eq!(TotalStakingAsset::<T>::get(), 200_000);
+        assert_eq!(TotalVoucher::<T>::get(), 5_000_000);
+
+        assert_last_event::<T>(Event::<T>::RewardsRecorded(agent, amount).into());
+    }
+
+    record_slash {
+        let caller: T::AccountId = whitelisted_caller();
+        initial_set_up::<T>(caller.clone());
+        let agent: T::AccountId = account("Sample", 6, SEED);
+        let amount = 100_000;
+        let slash_amount = 50_000;
+        assert_ok!(LiquidStaking::<T>::stake(
+            SystemOrigin::Signed(caller.clone()).into(),
+            amount));
+        let call = Call::<T>::record_slash(agent.clone(), slash_amount);
+        let origin = SystemOrigin::Root.into();
+    }: { call.dispatch_bypass_filter(origin)? }
+    verify {
+        assert_eq!(TotalStakingAsset::<T>::get(), slash_amount);
+        assert_eq!(TotalVoucher::<T>::get(), 5_000_000);
+
+        assert_last_event::<T>(Event::<T>::SlashRecorded(agent, slash_amount).into());
+    }
+
+
+    unstake {
+        let caller: T::AccountId = account("Sample", 1, SEED);
+        initial_set_up::<T>(caller.clone());
+        let amount = 100_000;
+        let unstake_amount = 5_000_000;
+        let exchange_rate = ExchangeRate::<T>::get();
+        let asset_amount = exchange_rate.checked_mul_int(unstake_amount).unwrap();
+        assert_ok!(LiquidStaking::<T>::stake(
+            SystemOrigin::Signed(caller.clone()).into(),
+            amount));
+    }: _(SystemOrigin::Signed(caller.clone()), unstake_amount)
+    verify {
+        assert_eq!(TotalStakingAsset::<T>::get(), 0);
+        assert_eq!(TotalVoucher::<T>::get(), 0);
+        assert_eq!(
+            <T as Config>::Currency::free_balance(T::StakingCurrency::get(), &caller),
+            INITIAL_AMOUNT - amount
+        );
+        assert_eq!(
+            <T as Config>::Currency::free_balance(T::StakingCurrency::get(), &LiquidStaking::<T>::account_id()),
+            INITIAL_AMOUNT + amount
+        );
+
+        assert_last_event::<T>(Event::<T>::Unstaked(caller, unstake_amount, asset_amount).into());
+    }
+
+    process_pending_unstake {
```

Comment:
`process_pending_unstake` and `finish_processed_unstake ` need to take consideration of `MaxAccountProcessingUnstake `.  The worst scenario is the max number of this constant, but it's possible to use this constant as a variable in the weight annotation.